TypeCobol/Compiler/Diagnostics/CodeElementWithTokensChecker.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Directives;
+using TypeCobol.Compiler.Parser;
+using TypeCobol.Compiler.Scanner;
+using TypeCobol.Compiler.Text;
+
+namespace TypeCobol.Compiler.Diagnostics
+{
+    class CodeElementWithTokensChecker
+    {
+
+        public CodeElementWithTokensChecker()
+        {
+
+        }
+
+        public static void CheckAreaOfDeclaration(CodeElement codeElement)
+        {
+            if (codeElement.ConsumedTokens == null)
+            {
+                return;
+            }
+            int currentLine = codeElement.Line;
+            foreach (var token in codeElement.ConsumedTokens)
```

Comment:
I made a mistake in the issue.
I've updated it :
- A CodeElement can be on multiple lines:
    - If  `StartingArea` is `A`, then only the first token of the CodeElement must start in A. Use `Line` and `LineEnd` to check if there are multiple lines. 
    - If  `StartingArea` is `B`, then allttokens of the CodeElement must start in B. Iterate over `ConsumedTokens` to check the column of first token of each line.

You also need to skip what we call FormalizedComment and MultilineComment.
See method [TypeCobol.LanguageServer.TypeCobolLanguageServer#CollectTokens](https://github.com/TypeCobolTeam/TypeCobol/blob/e830129f01afff13cba87793e19e21c1a93e1728/TypeCobol.LanguageServer/TypeCobolLanguageServer.cs#L154)

Reply:
The code you linked to is actually collecting the formalized/multline comment tokens instead of avoiding them. Also it looks complex for what it aims for. What about skipping tokens whose family is either FormalizedComment or MultilineComment ?

```csharp
            foreach (var token in codeElement.ConsumedTokens.SkipWhile(IsFormalizedOrMultilineCommentToken))
            {
                //...
            }

            bool IsFormalizedOrMultilineCommentToken(Token token)
            {
                return token.TokenFamily == TokenFamily.FormalizedCommentsFamily ||
                       token.TokenFamily == TokenFamily.MultilinesCommentsFamily;
            }
```

Reply:
Yes @fm-117 this is a much better code.

Reply:
Well it does not work actually... Because ':' is used in formalized comments but is not part of FormalizedCommentsFamily (and maybe other characters have the same problem). So it looks like we have to skip until we have found the end of formalized comments/multilines comments.

Reply:
```csharp
bool IsFormalizedOrMultilineCommentToken(Token token)
{
    var scanState = token.ScanStateSnapshot;
    if (scanState == null) return false;
    return scanState.InsideFormalizedComment || scanState.InsideMultilineComments || token.TokenType == TokenType.MULTILINES_COMMENTS_STOP || token.TokenType == TokenType.FORMALIZED_COMMENTS_STOP))
}
```

This needs to be tested though. I haven't checked.